File: src/libs/SspUis.BizLogicLayer/Hrm/Info/PartisanshipServices/Concrete/PartisanshipService.cs
using Microsoft.EntityFrameworkCore;
using SspUis.Core.Security;
using SspUis.DataLayer;
using SspUis.DataLayer.EfClasses.Hrm;
using SspUis.DataLayer.Repositories;
using System.Linq;
using WEBASE;
using WEBASE.Models;

namespace SspUis.BizLogicLayer.Hrm.PartisanshipServices.Concrete
{
    public class PartisanshipService : BaseEntityService<Partisanship, PartisanshipListDto, PartisanshipDto, CreatePartisanshipDlDto, UpdatePartisanshipDlDto, IPartisanshipRepository>, IPartisanshipService
    {
        public PartisanshipService(IUnitOfWork unitOfWork, IAuthService authService)
        : base(unitOfWork)
        {
        }

        public override PagedResult<PartisanshipListDto> GetList(SortFilterPageOptions dto)
        {
            var result = Repository.ReadAsNoTracked<PartisanshipListDto>().SortFilter(dto).AsPagedResult(dto);
            return result;
        }

        public override PartisanshipDto Get()
        {
            return new PartisanshipDto();
        }

        public override PartisanshipDto Get(int id)
        {
            var dto = Repository.ById<PartisanshipDto>(id);
            CombineStatuses(Repository);
            return dto;
        }

        public SelectList<int> AsSelectList()
        {
            return Repository.AllAsQueryable
                .AsSelectList();
        }

        public override HaveId<int> Create(CreatePartisanshipDlDto dto)
        {
            var entity = Repository.Create(dto, ent => Validation(dto, ent));
            CombineStatuses(Repository);
            if (IsValid)
            {
                UnitOfWork.Save();
                return HaveId.Create(entity.Id);
            }
            return null;
        }

        public override void Update(UpdatePartisanshipDlDto dto)
        {
            Repository.Update(dto, ent => Validation(dto, ent));
            CombineStatuses(Repository);
            if (IsValid)
                UnitOfWork.Save();
        }

        public override void Delete(int id)
        {
            try
            {
                Repository.Delete(id);
                CombineStatuses(Repository);
                if (IsValid)
                    UnitOfWork.Save();
            }
            catch (DbUpdateException)
            {
                AddError("Запись не может быть удален");
            }
        }

        private void Validation<TDto>(PartisanshipDlDto<TDto> dto, Partisanship entity)
            where TDto : PartisanshipDlDto<TDto>
        {
            var query = Repository.AllAsQueryable;

            if (entity != null)
                query = query.Where(a => a.Id != entity.Id);
        }
    }
}

File: ssp_uis_frontend/spp_uis_frontend_erp/src/services/hrm/partisanship.service.js
import ApiService from '../api.service';

// http://sspuis.apptest.uz/api/hrm/Partisanship/Create

const PartisanshipService = {
   GetList(data) {
      return ApiService.post('hrm/Partisanship/GetList', data);
   },
   Get(id) {
      if (id == 0 || id === null || id === undefined) {
         return ApiService.get('hrm/Partisanship/Get');
      } else {
         return ApiService.get(`hrm/Partisanship/Get/${id}`);
      }
   },
   Update(data) {
      if (data.id == 0) {
         return ApiService.post(`hrm/Partisanship/Create`, data);
      } else {
         return ApiService.post(`hrm/Partisanship/Update`, data);
      }
   },
   Delete(id) {
      return ApiService.post(`hrm/Partisanship/Delete/${id}`);
   },
   GetAsSelectList() {
      return ApiService.get(`hrm/Partisanship/GetAsSelectList`);
   }
};

export default PartisanshipService;
